app/code/Magento/ImportExport/Model/Import/Source/FileParser/CompositeParserFactory.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+/**
+ * File Parser Composite
+ *
+ */
+class CompositeParserFactory implements ParserFactoryInterface
+{
+    /** @var ParserFactoryInterface[] */
+    private $parserFactories;
+
+    public function __construct(array $parserFactories = [])
+    {
+        $this->parserFactories = [];
+
+        foreach ($parserFactories as $parserFactory) {
+            $this->addParserFactory($parserFactory);
+        }
```

Comment:
Variadic types are not supported by DI and I doubt it should be supported.
What is correct that DI should make a type validation for arrays, and it probably would do that in future.

For now, there are two possibilities to make a type check : 

1. Use xsi:type="object" not xsi:type="string". But doing that way you can't implement lazy loading, because ObjectManager will create an entity of needed type for you

2. Use fast type check in the constructor. It does not look well, but quite fast and adding comment that it's just for type check that would be easy to remove that in future when DI would be responsible for this operation
```php
// Type check for array to make sure 
// that all parser factories implement ParserFactoryInterface
$this->parserFactories = (function(ParserFactoryInterface ...$parserFactories) {
	return $parserFactories;
})(...$parserFactories);
```